### ISSUE TITLE
The provided cwd "" does not exist.

### ISSUE THREAD

**Issue by homersimpsons** [composer/composer#11217]
My `composer.json`:

```json
{
    "require": {
        "mouf/mouf": "^2.0"
    },
    "config": {
        "allow-plugins": {
            "mouf/mouf-installer": true
        }
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK git version 2.25.1
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 2.4.4
PHP version: 8.0.23
PHP binary path: /usr/bin/php8.0
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
cURL version: 7.68.0 libz 1.2.11 ssl OpenSSL/1.1.1f
zip: extension present, unzip present, 7-Zip not available
```

When I run this command: 

```
composer require mouf/mouf -vvv
```

I get the following output: 

```
Running 2.4.4 (2022-10-27 14:39:29) with PHP 8.0.23 on Linux / 5.15.0-53-generic
Reading ./composer.json (/tmp/b/composer.json)
Loading config file ./composer.json (/tmp/b/composer.json)
Checked CA file /etc/pki/tls/certs/ca-bundle.crt does not exist or it is not a file.
Checked directory /etc/pki/tls/certs/ca-bundle.crt does not exist or it is not a directory.
Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
Executing command (/tmp/b): 'git' 'branch' '-a' '--no-color' '--no-abbrev' '-v'
Executing command (/tmp/b): git describe --exact-match --tags
Executing command (CWD): git --version
Executing command (/tmp/b): git log --pretty="%H" -n1 HEAD --no-show-signature
Executing command (/tmp/b): hg branch
Executing command (/tmp/b): fossil branch list
Executing command (/tmp/b): fossil tag list
Executing command (/tmp/b): svn info --xml
Reading /home/docker/.composer/composer.json
Loading config file /home/docker/.composer/composer.json (/home/docker/.composer/composer.json)
Reading /tmp/b/vendor/composer/installed.json
Reading /home/docker/.composer/vendor/composer/installed.json
Loading plugin Mouf\Installer\MoufPlugin (from mouf/mouf-installer)
Reading /home/docker/.composer/cache/repo/https---repo.packagist.org/packages.json from cache
Downloading https://repo.packagist.org/packages.json if modified
[304] https://repo.packagist.org/packages.json
Reading /home/docker/.composer/cache/repo/https---repo.packagist.org/provider-mouf~mouf.json from cache
Downloading https://repo.packagist.org/p2/mouf/mouf.json if modified
[304] https://repo.packagist.org/p2/mouf/mouf.json
Using version ^2.0 for mouf/mouf
./composer.json has been updated
Unloading plugin Mouf\Installer\MoufPlugin
Reading ./composer.json (/tmp/b/composer.json)
Loading config file ./composer.json (/tmp/b/composer.json)
Executing command (/tmp/b): 'git' 'branch' '-a' '--no-color' '--no-abbrev' '-v'
Executing command (/tmp/b): git describe --exact-match --tags
Executing command (/tmp/b): git log --pretty="%H" -n1 HEAD --no-show-signature
Executing command (/tmp/b): hg branch
Executing command (/tmp/b): fossil branch list
Executing command (/tmp/b): fossil tag list
Executing command (/tmp/b): svn info --xml
Reading /home/docker/.composer/composer.json
Loading config file /home/docker/.composer/composer.json (/home/docker/.composer/composer.json)
Reading /tmp/b/vendor/composer/installed.json
Reading /home/docker/.composer/vendor/composer/installed.json
Loading plugin Mouf\Installer\MoufPlugin_composer_tmp0 (from mouf/mouf-installer)
Running composer update mouf/mouf
Reading ./composer.lock (/tmp/b/composer.lock)
Loading composer repositories with package information
Reading /home/docker/.composer/cache/repo/https---repo.packagist.org/packages.json from cache
Downloading https://repo.packagist.org/packages.json if modified
[304] https://repo.packagist.org/packages.json
Reading /home/docker/.composer/cache/repo/https---repo.packagist.org/provider-mouf~mouf.json from cache
Downloading https://repo.packagist.org/p2/mouf/mouf.json if modified
[304] https://repo.packagist.org/p2/mouf/mouf.json
Built pool.
Running pool optimizer.
Pool optimizer completed in 0.014 seconds
Found 125 package versions referenced in your dependency graph. 38 (30%) were optimized away.
Updating dependencies
Generating rules
Resolving dependencies through SAT
Looking at all rules.

Dependency resolution completed in 0.001 seconds
Analyzed 87 packages to resolve dependencies
Analyzed 109 rules to resolve dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
Installs: mouf/mouf:v2.0.42
Reading /home/docker/.composer/cache/files/mouf/mouf/1559a0e257dc804d58a5d2fd13fbd8bb6b763e0f.zip from cache
  - Loading mouf/mouf (v2.0.42) from cache
  - Installing mouf/mouf (v2.0.42): Extracting archive
Executing command (CWD): rm -rf '/tmp/b/vendor/mouf/mouf/doc'
Executing command (CWD): rm -rf '/tmp/b/vendor/mouf/mouf/mouf'
Executing command (CWD): rm -rf '/tmp/b/vendor/mouf/mouf/tests'
Executing command (CWD): rm -rf '/tmp/b/vendor/mouf/mouf/src-dev'
Executing command (CWD): rm -rf '/tmp/b/vendor/mouf/mouf/src'
Executing async command (CWD): '/usr/bin/unzip' -qq '/tmp/b/vendor/composer/tmp-736961cb34963ad82471e035533b2790' -d '/tmp/b/vendor/composer/8556de2e'
    Install of mouf/mouf failed
Executing async command (CWD): rm -rf '/tmp/b/vendor/composer/8556de2e'
Executing command (CWD): rm -rf '/tmp/b/vendor/composer/8556de2e'

In Process.php line 349:
                                                          
  [Symfony\Component\Process\Exception\RuntimeException]  
  The provided cwd "" does not exist.                     
                                                          

Exception trace:
  at phar:///usr/local/bin/composer/vendor/symfony/process/Process.php:349
 Symfony\Component\Process\Process->start() at phar:///usr/local/bin/composer/vendor/symfony/process/Process.php:252
 Symfony\Component\Process\Process->run() at phar:///usr/local/bin/composer/src/Composer/Util/ProcessExecutor.php:134
 Composer\Util\ProcessExecutor->doExecute() at phar:///usr/local/bin/composer/src/Composer/Util/ProcessExecutor.php:76
 Composer\Util\ProcessExecutor->execute() at phar:///usr/local/bin/composer/src/Composer/Util/Filesystem.php:116
 Composer\Util\Filesystem->removeDirectory() at phar:///usr/local/bin/composer/src/Composer/Util/Filesystem.php:42
 Composer\Util\Filesystem->remove() at phar:///usr/local/bin/composer/src/Composer/Downloader/FileDownloader.php:330
 Composer\Downloader\FileDownloader->cleanup() at phar:///usr/local/bin/composer/src/Composer/Downloader/ArchiveDownloader.php:46
 Composer\Downloader\ArchiveDownloader->cleanup() at phar:///usr/local/bin/composer/src/Composer/Downloader/DownloadManager.php:356
 Composer\Downloader\DownloadManager->cleanup() at phar:///usr/local/bin/composer/src/Composer/Installer/LibraryInstaller.php:136
 Composer\Installer\LibraryInstaller->cleanup() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:293
 Composer\Installer\InstallationManager::Composer\Installer\{closure}() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:194
 Composer\Installer\InstallationManager::Composer\Installer\{closure}() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:245
 React\Promise\Promise->call() at phar:///usr/local/bin/composer/vendor/react/promise/src/Promise.php:25
 React\Promise\Promise->__construct() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:202
 Composer\Installer\InstallationManager::Composer\Installer\{closure}() at phar:///usr/local/bin/composer/src/Composer/Installer/InstallationManager.php:245
 Composer\Installer\InstallationManager->execute() at phar:///usr/local/bin/composer/src/Composer/Installer.php:786
 Composer\Installer->doInstall() at phar:///usr/local/bin/composer/src/Composer/Installer.php:616
 Composer\Installer->doUpdate() at phar:///usr/local/bin/composer/src/Composer/Installer.php:281
 Composer\Installer->run() at phar:///usr/local/bin/composer/src/Composer/Command/RequireCommand.php:477
 Composer\Command\RequireCommand->doUpdate() at phar:///usr/local/bin/composer/src/Composer/Command/RequireCommand.php:331
 Composer\Command\RequireCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:1028
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:375
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:139
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:88
 require() at /usr/local/bin/composer:29

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>...]
```

And I expected this to happen:

The command should run smoothly. without the 
```
[Symfony\Component\Process\Exception\RuntimeException]  
  The provided cwd "" does not exist. 
```

## Diagnostic

Looking at the stack trace the issue seems to come from composer:

1. We tries to remove a directory
https://github.com/composer/composer/blob/e8d9087229bcdbc5867594d3098091412f1130cf/src/Composer/Util/Filesystem.php#L116

2. This call then calls the `ProcessExecutor::doExecute` with `$cwd = null`
https://github.com/composer/composer/blob/e8d9087229bcdbc5867594d3098091412f1130cf/src/Composer/Util/ProcessExecutor.php#L76

4. Which then creates a `Symfony\Component\Process\Process` with a `$cwd = null`
https://github.com/composer/composer/blob/e8d9087229bcdbc5867594d3098091412f1130cf/src/Composer/Util/ProcessExecutor.php#L110

5. This goes then to Symfony internals where it does not get replaced because thread safety is disabled
https://github.com/symfony/symfony/blob/3b75e0ac5af1f765951d4978ea02ed491aeb81b3/src/Symfony/Component/Process/Process.php#L156

6. This ultimately throws the exception
https://github.com/symfony/symfony/blob/3b75e0ac5af1f765951d4978ea02ed491aeb81b3/src/Symfony/Component/Process/Process.php#L348

### Thread safety

I do not know if this should be enabled or not, but when I check it is diasbled (on php 8.1.2):

```
❯ php -i | grep Thread
Thread Safety => disabled
```

## Additional notes

Note that I am a maintainer on https://github.com/thecodingmachine/mouf-installer and https://github.com/thecodingmachine/mouf so if anything is wrong with our composer plugin I can fix it. Installing without the plugin leads to no error, but to me the error in the plugin triggers the error in composer but there still is an issue in composer.

```
composer require mouf/mouf --no-plugins
```

**Comment by Seldaek** [comment]
Is it really due to thread safety? Can you check if you do force the ` $this->cwd = getcwd();` if that fixes it?

Or is it because the mouf-installer perhaps does a chdir() into a directory which is then wiped or something? Or perhaps it's a bug in Composer doing the same but I kinda doubt it as it's first time I see this.

**Comment by homersimpsons** [comment]
> Is it really due to thread safety?

This is a guess because of this line from `symfony/process`:

```php
if (null === $this->cwd && (\defined('ZEND_THREAD_SAFE') || '\\' === \DIRECTORY_SEPARATOR)) {
```

> Can you check if you do force the $this->cwd = getcwd(); if that fixes it?

@Seldaek How can I do that as composer is just a phar ?

> Or is it because the mouf-installer perhaps does a chdir() into a directory which is then wiped or something?

This is possible

> Or perhaps it's a bug in Composer doing the same but I kinda doubt it as it's first time I see this.

I could not find any mention of this elsewhere, too. The generic `The provided cwd "" does not exist.` has been reported on other projects (such as laravel if I remember) but do not look related.

**Comment by Seldaek** [comment]
> @Seldaek How can I do that as composer is just a phar ?

You can clone this repo and composer install in it and then run bin/composer isntead of the phar, then you can go edit stuff in the vendor dir if needed.

**Comment by homersimpsons** [comment]
Thank you for the tip. I'm investigating currently, and I know what is happening, but I do not know why nor how.

I did reproduce the issue with `strace` and I can see the following output:

```sh
> cd /tmp
> strace --follow-forks -e rmdir,chdir,getcwd composer require mouf/mouf
[...]
[pid 54112] chdir("/tmp")               = 0
[pid 54074] getcwd("/tmp", 4096)        = 5
[pid 54074] chdir("vendor/mouf/mouf")   = 0
[pid 54074] getcwd("/tmp/vendor/mouf/mouf", 4096) = 22
    Install of mouf/mouf failed
strace: Process 54113 attached
[pid 54113] +++ exited with 0 +++
[pid 54112] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54113, si_uid=1000, si_status=0, si_utime=5, si_stime=2} ---
[pid 54112] +++ exited with 0 +++
[pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54112, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
[pid 54074] rmdir("/tmp/vendor/mouf/mouf") = 0
[pid 54074] getcwd(0x7ffccebf45d0, 4096) = -1 ENOENT (No such file or directory)
[pid 54074] getcwd(0x7ffccebf4ef0, 4096) = -1 ENOENT (No such file or directory)
[...]
```

When I look at the `strace` I see that for the process `54074` there is a successful `getcwd` call but then it receives the `SIGCHLD` signal, after this signal it `rmdir` the directory he is currently in and so the subsequent `getcwd` fails.

The process changes its directory with the `chdir("vendor/mouf/mouf")` call. This call may be part of the `mouf/mouf` or `mouf/mouf-installer` code, I am looking for this.

<details>
  <summary>Full output of the above sample</summary>

  ```sh
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  getcwd("/tmp", 4096)                    = 5
  strace: Process 54074 attached
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54075 attached
  strace: Process 54076 attached
  strace: Process 54077 attached
  [pid 54076] +++ exited with 0 +++
  [pid 54075] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54076, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54077] +++ exited with 0 +++
  [pid 54075] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54077, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54075] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54075, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  strace: Process 54078 attached
  strace: Process 54079 attached
  [pid 54079] +++ exited with 1 +++
  [pid 54078] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54079, si_uid=1000, si_status=1, si_utime=0, si_stime=0} ---
  [pid 54078] +++ exited with 1 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54078, si_uid=1000, si_status=1, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54080 attached
  [pid 54080] chdir("/tmp")               = 0
  [pid 54080] getcwd("/tmp", 129)         = 5
  [pid 54080] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54080, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  strace: Process 54081 attached
  [pid 54081] chdir("/tmp")               = 0
  strace: Process 54082 attached
  [pid 54082] getcwd("/tmp", 129)         = 5
  [pid 54082] +++ exited with 128 +++
  [pid 54081] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54082, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  [pid 54081] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54081, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54083 attached
  [pid 54083] chdir("/tmp")               = 0
  strace: Process 54084 attached
  [pid 54084] +++ exited with 0 +++
  [pid 54083] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54084, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54083] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54083, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  strace: Process 54085 attached
  [pid 54085] chdir("/tmp")               = 0
  strace: Process 54086 attached
  [pid 54086] getcwd("/tmp", 129)         = 5
  [pid 54086] +++ exited with 128 +++
  [pid 54085] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54086, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  [pid 54085] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54085, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  strace: Process 54087 attached
  [pid 54087] chdir("/tmp")               = 0
  [pid 54087] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54087, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54088 attached
  [pid 54088] chdir("/tmp")               = 0
  [pid 54088] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54088, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54089 attached
  [pid 54089] chdir("/tmp")               = 0
  [pid 54089] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54089, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54090 attached
  [pid 54090] chdir("/tmp")               = 0
  [pid 54090] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54090, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54091 attached
  [pid 54091] +++ exited with 0 +++
  Info from https://repo.packagist.org: #StandWithUkraine
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  ./composer.json has been updated
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54092 attached
  [pid 54092] chdir("/tmp")               = 0
  [pid 54092] getcwd("/tmp", 129)         = 5
  [pid 54092] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54092, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  strace: Process 54093 attached
  [pid 54093] chdir("/tmp")               = 0
  strace: Process 54094 attached
  [pid 54094] getcwd("/tmp", 129)         = 5
  [pid 54094] +++ exited with 128 +++
  [pid 54093] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54094, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  [pid 54093] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54093, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  strace: Process 54095 attached
  [pid 54095] chdir("/tmp")               = 0
  strace: Process 54096 attached
  [pid 54096] getcwd("/tmp", 129)         = 5
  [pid 54096] +++ exited with 128 +++
  [pid 54095] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54096, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  [pid 54095] +++ exited with 128 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54095, si_uid=1000, si_status=128, si_utime=0, si_stime=0} ---
  strace: Process 54097 attached
  [pid 54097] chdir("/tmp")               = 0
  [pid 54097] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54097, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54098 attached
  [pid 54098] chdir("/tmp")               = 0
  [pid 54098] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54098, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54099 attached
  [pid 54099] chdir("/tmp")               = 0
  [pid 54099] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54099, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  strace: Process 54100 attached
  [pid 54100] chdir("/tmp")               = 0
  [pid 54100] +++ exited with 127 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54100, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  Running composer update mouf/mouf
  [pid 54074] getcwd("/tmp", 4096)        = 5
  Loading composer repositories with package information
  strace: Process 54101 attached
  [pid 54101] +++ exited with 0 +++
  Updating dependencies
  Nothing to modify in lock file
  Installing dependencies from lock file (including require-dev)
  Package operations: 1 install, 0 updates, 0 removals
    - Installing mouf/mouf (v2.0.44): Extracting archive
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54102 attached
  [pid 54102] chdir("/tmp")               = 0
  strace: Process 54103 attached
  [pid 54103] +++ exited with 0 +++
  [pid 54102] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54103, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54102] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54102, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54104 attached
  [pid 54104] chdir("/tmp")               = 0
  strace: Process 54105 attached
  [pid 54105] +++ exited with 0 +++
  [pid 54104] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54105, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54104] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54104, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54106 attached
  [pid 54106] chdir("/tmp")               = 0
  strace: Process 54107 attached
  [pid 54107] +++ exited with 0 +++
  [pid 54106] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54107, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54106] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54106, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54108 attached
  [pid 54108] chdir("/tmp")               = 0
  strace: Process 54109 attached
  [pid 54109] +++ exited with 0 +++
  [pid 54108] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54109, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54108] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54108, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54110 attached
  [pid 54110] chdir("/tmp")               = 0
  strace: Process 54111 attached
  [pid 54111] +++ exited with 0 +++
  [pid 54110] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54111, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54110] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54110, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] getcwd("/tmp", 4096)        = 5
  strace: Process 54112 attached
  [pid 54112] chdir("/tmp")               = 0
  [pid 54074] getcwd("/tmp", 4096)        = 5
  [pid 54074] chdir("vendor/mouf/mouf")   = 0
  [pid 54074] getcwd("/tmp/vendor/mouf/mouf", 4096) = 22
      Install of mouf/mouf failed
  strace: Process 54113 attached
  [pid 54113] +++ exited with 0 +++
  [pid 54112] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54113, si_uid=1000, si_status=0, si_utime=5, si_stime=2} ---
  [pid 54112] +++ exited with 0 +++
  [pid 54074] --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54112, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
  [pid 54074] rmdir("/tmp/vendor/mouf/mouf") = 0
  [pid 54074] getcwd(0x7ffccebf45d0, 4096) = -1 ENOENT (No such file or directory)
  [pid 54074] getcwd(0x7ffccebf4ef0, 4096) = -1 ENOENT (No such file or directory)
  
  In Process.php line 349:
                                         
    The provided cwd "" does not exist.  
                                         
  
  require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>...]
  
  [pid 54074] +++ exited with 1 +++
  --- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=54074, si_uid=1000, si_status=1, si_utime=43, si_stime=12} ---
  +++ exited with 1 +++
  ```
</details>

**Comment by homersimpsons** [comment]
The `chdir` clearly is there:

https://github.com/thecodingmachine/mouf-installer/blob/7b251ace85dae31ffd1723df7d74c2b3207fdf6e/src/Mouf/Installer/MoufFrameworkInstaller.php#L52

I think I found the issue. Basically `mouf/mouf` install itself in its own vendor directory thanks to the above Composer Plugin.

The issue is that `mouf/mouf` requires `mouf/html.utils.weblibrarymanager` which requires `mouf/html.utils.weblibrarymanager.component-installer` which requires `composer-plugin-api 1.0.0` and that conflicts with `composer-plugin-api[2.3.0]`.

Thank you for the suggestion to clone this repository, in fact I was able to try things and understand better how the `composer install` works.
